### ISSUE TITLE
While loops and accompanying features

### ISSUE THREAD

**Issue by Ashvin-Ranjan** [nbuilding/N-lang#154]
## While loops
An important feature missing from N for too long has been the `while` loop. While it may not have too many useful functionalities, now that the `for` loop has started iterating through lists, it is important that this feature is to be added. There are many ways of implementing it but I will currently suggest this:
```js
import SystemIO
let name = ""
while (~validName(name)) {
  var name = SystemIO.inp("Please input a name")
}
```
This would allow for easy usage of the `while` loop that is familiar to those that do not know N that well, as it imitates the way that other programming languages use `while` loops.
## Continue
`continue` is a keyword already reserved in the fixing of #80 along with `while` and `break`. It would be added in a way that one can just call it like a simple statement as such:
```js
for (i in range(0, 10, 1)) {
  if (i == 1) {
    continue
  }
  print(i)
}
```
This would be a simple, and familiar way of using `continue` to most people. It should be easy to implement as it is similar to `return` but with less error checking.
## Break
`break` is a very important keyword that for some reason has been left unimplemented for a long time, yet it reserved as a keyword when #80 was fixed. The way `break` would work is that it would be called in a similar way to `continue` except instead of continuing to the next iteration, it would break out of the loop entirely, this could also be used in `if` statements as it might be very helpful. There are some foreseeable issues with scoping break as in the situation.
```js
if (1 == 1) {
  if (true) {
    break
  }
  if (true) {
   print("hi")
  }
}
```
it may be that the user does not want to exit out of the first if statement of vice versa, in this situation it is hard to accurately judge the scope of the `break`.

**Comment by Ashvin-Ranjan** [comment]
For
```js
let wow = if something > 3 {
  break
} else {
  2 + whatever
}
```
Would that make `wow` a `maybe[int]`?

**Comment by Ashvin-Ranjan** [comment]
Closed as of #164
